website/docs/tutorials/guides/testing.md

```diff
@@ -443,18 +443,68 @@ Deploys a contract given a path relative to a Protostar project root. The sectio
 ### `start_prank`
 
 ```python
-def start_prank(caller_address: int) -> None: ...
+def start_prank(caller_address: int, target_contract_addr: Optional[int] = None) -> Callable: ...
 ```
 
-Changes caller address until [`stop_prank`](#stop_prank) cheatcode is used.
+Changes caller address until the returned callable is called. If `target_contract_addr` specified, prank affects only the contract with the specified address. Otherwise prank affects the current contract.
 
-### `stop_prank`
+#### With target
+```cairo title="./pranked_contract.cairo"
+%lang starknet
 
-```python
-def stop_prank() -> None: ...
+from starkware.starknet.common.syscalls import (get_caller_address)
+
+@view
+func assert_pranked{syscall_ptr : felt*}():
+    let (caller_addr) = get_caller_address()
+    with_attr error_message("Not pranked!"):
+        assert caller_addr = 123
+    end
+    return ()
+end
 ```
+```cairo title="Remote assert passes"
+@contract_interface
+namespace Pranked:
+    func assert_pranked() -> ():
+    end
+end
+
+@external
```

Comment:
Maybe we should stick to `@view` everywhere.

Reply:
I think we probably should stick to `@external `, cairo docs:

> The @view decorator is identical to the @external decorator. The only difference is that the method is annotated as a method that only queries the state rather than modifying it. Note that in the current version this is not enforced by the compiler.

Reply:
https://github.com/software-mansion/protostar/issues/272

Reply:
Nothing, it's just docs